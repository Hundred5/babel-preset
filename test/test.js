const babel = require("@babel/core");
const path = require("path");
const util = require("util");

const transformFile = util.promisify(babel.transformFile);
const preset = path.join(__dirname, "..", "index.js");

it("transforms for the browser environment", async () => {
  const result = await transformFile(path.join(__dirname, "file.tsx"), {
    presets: [preset],
    envName: "development",
    cwd: __dirname
  });

  expect(result.code).toMatchSnapshot();
});

it("transforms for the test environment", async () => {
  const result = await transformFile(path.join(__dirname, "file.tsx"), {
    presets: [preset],
    envName: "test"
  });

  expect(result.code).toMatchSnapshot();
});
