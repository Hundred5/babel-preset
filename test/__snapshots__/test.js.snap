// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transforms for the browser environment 1`] = `
"import _regeneratorRuntime from \\"@babel/runtime/regenerator\\";
import _defineProperty from \\"@babel/runtime/helpers/defineProperty\\";

var _ref;

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(source, true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(source).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var x = import(\\"x\\");
import \\"y\\";
var element = React.createElement(\\"div\\", null, \\"Hello!\\");
var jobOpeningOne = {
  title: \\"Developer\\"
};

var jobOpeningTwo = _objectSpread({}, jobOpeningOne);

var description = (_ref = jobOpeningTwo === null || jobOpeningTwo === void 0 ? void 0 : jobOpeningTwo.description) !== null && _ref !== void 0 ? _ref : \\"Default\\";

var fn = function fn() {
  return _regeneratorRuntime.async(function fn$(_context) {
    while (1) {
      switch (_context.prev = _context.next) {
        case 0:
          return _context.abrupt(\\"return\\", true);

        case 1:
        case \\"end\\":
          return _context.stop();
      }
    }
  });
};"
`;

exports[`transforms for the test environment 1`] = `
"\\"use strict\\";

var _interopRequireDefault = require(\\"@babel/runtime/helpers/interopRequireDefault\\");

var _defineProperty2 = _interopRequireDefault(require(\\"@babel/runtime/helpers/defineProperty\\"));

var _interopRequireWildcard2 = _interopRequireDefault(require(\\"@babel/runtime/helpers/interopRequireWildcard\\"));

require(\\"y\\");

var _ref;

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(source, true).forEach(function (key) { (0, _defineProperty2.default)(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(source).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

const x = Promise.resolve().then(() => (0, _interopRequireWildcard2.default)(require(\\"x\\")));
const element = React.createElement(\\"div\\", null, \\"Hello!\\");
const jobOpeningOne = {
  title: \\"Developer\\"
};

const jobOpeningTwo = _objectSpread({}, jobOpeningOne);

const description = (_ref = jobOpeningTwo === null || jobOpeningTwo === void 0 ? void 0 : jobOpeningTwo.description) !== null && _ref !== void 0 ? _ref : \\"Default\\";

const fn = async () => {
  return true;
};"
`;
